File: day02/py/day02.py
#Read data 
text_file = open("../input.txt", "r")
lines = [v.strip("\n").split() for v in text_file.readlines()]

def p1():
    mydict = {"horizontal":0,"depth":0}
    for line in lines:
        if line[0]=="forward":
            mydict["horizontal"]+=int(line[1])
        elif line[0]=="down":
            mydict["depth"]+=int(line[1])
        elif line[0]=="up":
            mydict["depth"]-=int(line[1])

    print("Part 1:",mydict["horizontal"]*mydict["depth"])

def p2():
    mydict = {"horizontal":0,"depth":0,"aim":0}
    for line in lines:
        if line[0]=="forward":
            mydict["horizontal"]+=int(line[1])
            mydict["depth"]+=int(line[1])*mydict["aim"]
        elif line[0]=="down":
            mydict["aim"]+=int(line[1])
        elif line[0]=="up":
            mydict["aim"]-=int(line[1])

    print("Part 2:",mydict["horizontal"]*mydict["depth"])

p1()
p2()

File: day02/lua/day02.lua
-- stolen from here: https://stackoverflow.com/questions/11201262/how-to-read-data-from-a-file-in-lua
function lines_from(file)
    lines = {}
    for line in io.lines(file) do 
      lines[#lines + 1] = line
    end
    return lines
end

-- Stolen from here: https://stackoverflow.com/questions/1426954/split-string-in-lua
function mysplit (inputstr, sep)
    if sep == nil then
            sep = "%s"
    end
    local t={}
    for str in string.gmatch(inputstr, "([^"..sep.."]+)") do
            table.insert(t, str)
    end
    return t
end

-- Read the data
local file = '../input.txt'
local lines = lines_from(file)
  
function p1 (lines)
    local horizontal = 0
    local depth = 0

    for k,v in pairs(lines) do
        instrux=mysplit(v," ")[1]
        thenum=tonumber(mysplit(v," ")[2])
        -- print(k,instrux,thenum)
        if instrux == "forward" then
            horizontal=horizontal+thenum
        elseif instrux == "down" then
            depth = depth+thenum
        elseif instrux == "up" then
            depth = depth-thenum
        end
    end
    print("Final Answer:",horizontal*depth)
end

function p2 (lines)
    local horizontal = 0
    local depth = 0
    local aim = 0

    for k,v in pairs(lines) do
        instrux=mysplit(v," ")[1]
        thenum=tonumber(mysplit(v," ")[2])
        -- print(k,instrux,thenum)
        if instrux == "forward" then
            horizontal=horizontal+thenum
            depth = depth+aim*thenum
        elseif instrux == "down" then
            aim = aim+thenum
        elseif instrux == "up" then
            aim = aim-thenum
        end
    end
    print("Final Answer:",horizontal*depth)
end

p1(lines)
p2(lines)

